atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleAlterTableIntegrationTest.java

```diff
@@ -0,0 +1,176 @@
+        assertThatThrownBy(() -> fetchData(defaultKvs, DEFAULT_CELL_1, TIMESTAMP_1));
+
+        KeyValueService workingKvs = ConnectionManagerAwareDbKvs.create(CONFIG_WITH_ALTER);
+        workingKvs.createTable(TABLE_REFERENCE, EXPECTED_TABLE_METADATA.persistToBytes());
+        assertThatDataCanBeRead(defaultKvs, DEFAULT_CELL_1, TIMESTAMP_1);
+        assertThatOverflowColumnExists();
+        assertThatCode(() -> writeData(defaultKvs, ROW_2, TIMESTAMP_2)).doesNotThrowAnyException();
+        assertThatDataCanBeRead(defaultKvs, DEFAULT_CELL_2, TIMESTAMP_2);
+    }
+
+    @Test
+    public void whenConfiguredAlterTableDoesNothingWhenMatching() {
+        KeyValueService kvsWithAlter = ConnectionManagerAwareDbKvs.create(CONFIG_WITH_ALTER);
+        kvsWithAlter.createTable(TABLE_REFERENCE, EXPECTED_TABLE_METADATA.persistToBytes());
+        writeData(defaultKvs, ROW_1, TIMESTAMP_1);
+        assertThatDataCanBeRead(defaultKvs, DEFAULT_CELL_1, TIMESTAMP_1);
+
+        kvsWithAlter.createTable(TABLE_REFERENCE, EXPECTED_TABLE_METADATA.persistToBytes());
+        writeData(defaultKvs, ROW_2, TIMESTAMP_2);
+        assertThatDataCanBeRead(defaultKvs, DEFAULT_CELL_2, TIMESTAMP_2);
+    }
+
+    private void assertThatOverflowColumnExists() {
+        assertThatCode(() -> {
+                    String tableName =
+                            oracleTableNameGetter.getInternalShortTableName(connectionSupplier, TABLE_REFERENCE);
+                    boolean columnExists = connectionSupplier
+                            .get()
+                            .selectExistsUnregisteredQuery(
+                                    "SELECT 1 FROM user_tab_cols WHERE TABLE_NAME = ? AND COLUMN_NAME = 'OVERFLOW'",
+                                    tableName.toUpperCase());
+                    assertThat(columnExists).isTrue();
+                })
+                .doesNotThrowAnyException();
+    }
+
+    private void dropOverflowColumn() {
+        try {
+            String tableName = oracleTableNameGetter.getInternalShortTableName(connectionSupplier, TABLE_REFERENCE);
+            connectionSupplier.get().executeUnregisteredQuery("ALTER TABLE " + tableName + " DROP COLUMN overflow");
+        } catch (TableMappingNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void writeData(KeyValueService kvs, Map<Cell, byte[]> value, long timestamp) {
+        kvs.put(TABLE_REFERENCE, value, timestamp);
+    }
+
+    private static Map<Cell, Value> fetchData(KeyValueService kvs, Cell cell, long timestamp) {
+        return kvs.get(TABLE_REFERENCE, Map.of(cell, timestamp + 1));
+    }
+
+    private static void assertThatDataCanBeRead(KeyValueService kvs, Cell cell, long timestamp) {
+        Map<Cell, Value> values = fetchData(kvs, cell, timestamp);
+        assertThat(values).hasSize(1);
+        assertThat(values).containsKey(DEFAULT_CELL_1);
```

Comment:
I'm surprised this is working (or, we haven't actually ran the tests...), especially since you pass in CELL2 in places, but then assert against CELL_1